src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1286,7 +1286,8 @@ abstract class RefChecks extends Transform {
           reporter.warning(pos, s"${sym.fullLocationString} has changed semantics in version ${sym.migrationVersion.get}:\n${sym.migrationMessage.get}")
       }
       // See an explanation of compileTimeOnly in its scaladoc at scala.annotation.compileTimeOnly.
-      if (sym.isCompileTimeOnly && !currentOwner.ownerChain.exists(x => x.isCompileTimeOnly)) {
+      // async/await is expanded after erasure
+      if (sym.isCompileTimeOnly && sym != global.async.asyncSymbols.Async_await && !currentOwner.ownerChain.exists(x => x.isCompileTimeOnly)) {
```

Comment:
This should check for all registered `await` symbols, not just the default one. 3rd-party `async` macros will need the same exception for their own `await`s.